src/lib/journey/stages/generic.svelte

```diff
@@ -1,37 +1,69 @@
 <script lang="ts">
   import type { FRLoginFailure, FRLoginSuccess, FRStep } from '@forgerock/javascript-sdk';
 
+  // i18n
+  import { interpolate } from '$lib/utilities/i18n.utilities';
+  import T from '$components/i18n/index.svelte';
+
   // Import primitives
+  import Alert from '$components/primitives/alert/alert.svelte';
   import Button from '$components/primitives/button/button.svelte';
+  import { convertStringToKey } from '$journey/utilities/callback.utilities';
   import Form from '$components/primitives/form/form.svelte';
-  import T from '$components/i18n/index.svelte';
   import { mapCallbackToComponent } from '$journey/utilities/map-callback.utilities';
   import Spinner from '$components/primitives/spinner/spinner.svelte';
 
   type StepTypes = FRStep | FRLoginSuccess | FRLoginFailure | null;
 
+  export let failureMessage: string;
   export let formEl: HTMLFormElement | null = null;
   export let step: StepTypes;
   export let submitForm: () => void;
+
+  let failureMessageKey = '';
+  let hasPrevError = false;
+
+  // TODO: Pull out and rework into a utility or helper
+  function checkValidation(callback: any) {
+    let failedPolices = callback.getOutputByName('failedPolicies', []);
+    if (failedPolices.length && !hasPrevError) {
+      console.log(callback);
+      hasPrevError = true;
+      return true;
+    }
+    return false;
+  }
+
```

Comment:
You may see a few of these comments saying things should be moved to utility functions. There's some duplication of logic across components. The fact that there are some closures dependencies makes it a bit harder to move them out.